google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -167,23 +168,29 @@ public Project getProject() {
       return Optional.of(createStandardRunner(loggingHandler, Paths.get(source.getFilePath()),
           deploy, isFlexCompat));
     } else if (targetEnvironment.isFlexible()) {
-      // Checks if the Yaml or Dockerfile exist.
-      Optional<FlexibleRuntime> runtimeOptional =
-          AppEngineProjectService.getInstance().getFlexibleRuntimeFromAppYaml(
-              deploymentConfiguration.getAppYamlPath());
-
-      if (!Files.exists(Paths.get(deploymentConfiguration.getAppYamlPath()))) {
-        callback.errorOccurred(GctBundle.getString("appengine.deployment.error.staging.yaml"));
-        return Optional.empty();
-      }
-      if (runtimeOptional.filter(runtime -> runtime == FlexibleRuntime.custom).isPresent()
-          && !Files.exists(Paths.get(deploymentConfiguration.getDockerFilePath()))) {
+      try {
+        // Checks if the Yaml or Dockerfile exist.
+        Optional<FlexibleRuntime> runtimeOptional =
+            AppEngineProjectService.getInstance().getFlexibleRuntimeFromAppYaml(
+                deploymentConfiguration.getAppYamlPath());
+
+        if (!Files.exists(Paths.get(deploymentConfiguration.getAppYamlPath()))) {
+          callback.errorOccurred(GctBundle.getString("appengine.deployment.error.staging.yaml"));
+          return Optional.empty();
+        }
+        if (runtimeOptional.filter(runtime -> runtime == FlexibleRuntime.custom).isPresent()
+            && !Files.exists(Paths.get(deploymentConfiguration.getDockerFilePath()))) {
+          callback.errorOccurred(
+              GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+          return Optional.empty();
+        }
+        return Optional.of(createFlexRunner(loggingHandler, Paths.get(source.getFilePath()),
+            deploymentConfiguration, deploy));
+      } catch (MalformedYamlFileException myf) {
         callback.errorOccurred(
-            GctBundle.getString("appengine.deployment.error.staging.dockerfile"));
+            GctBundle.message("appengine.appyaml.malformed") + "\n" + myf.getMessage());
```

Comment:
Where does this get shown? Just confirming because I see you added the exception message here too.